cmd/debug.go

```diff
@@ -0,0 +1,176 @@
+					fmt.Printf("%s [%s]: %s\n", update.ServiceName, update.Status, line)
+				}
+			}
+		} else {
+			prog := teax.NewProgram(build.NewModel(buildUpdates, "Building Services"))
+			// blocks but quits once the above updates channel is closed by the build process
+			buildModel, err := prog.Run()
+			tui.CheckErr(err)
+			if buildModel.(build.Model).Err != nil {
+				tui.CheckErr(fmt.Errorf("error building services"))
+			}
+		}
+
+		// Step 2. Start the collectors and containers (respectively in pairs)
+		// Step 3. Merge requirements from collectors into a specification
+		serviceRequirements, err := proj.CollectServicesRequirements()
+		tui.CheckErr(err)
+
+		additionalEnvFiles := []string{}
+
+		if debugEnvFile != "" {
+			additionalEnvFiles = append(additionalEnvFiles, envFile)
+		}
+
+		envVariables, err := env.ReadLocalEnv(additionalEnvFiles...)
+		if err != nil && os.IsNotExist(err) {
+			if !os.IsNotExist(err) {
+				tui.CheckErr(err)
+			}
+			// If it doesn't exist set blank
+			envVariables = map[string]string{}
+		}
+
+		defaultImageName, ok := proj.DefaultMigrationImage(fs)
+		if !ok {
+			defaultImageName = ""
+		}
+
+		migrationImageContexts, err := collector.GetMigrationImageBuildContexts(serviceRequirements, fs)
+		tui.CheckErr(err)
+		// Build images from contexts and provide updates on the builds
+
+		if len(migrationImageContexts) > 0 {
+			migrationBuildUpdates, err := project.BuildMigrationImages(fs, migrationImageContexts)
+			tui.CheckErr(err)
+
+			if isNonInteractive() {
+				fmt.Println("building project migration images")
+				// non-interactive environment
+				for update := range migrationBuildUpdates {
+					for _, line := range strings.Split(strings.TrimSuffix(update.Message, "\n"), "\n") {
+						fmt.Printf("%s [%s]: %s\n", update.ServiceName, update.Status, line)
+					}
+				}
+			} else {
+				prog := teax.NewProgram(build.NewModel(migrationBuildUpdates, "Building Database Migrations"))
+				// blocks but quits once the above updates channel is closed by the build process
+				buildModel, err := prog.Run()
+				tui.CheckErr(err)
+				if buildModel.(build.Model).Err != nil {
+					tui.CheckErr(fmt.Errorf("error building services"))
+				}
+			}
+		}
+
+		outputFile := debugFile
+		if outputFile == "" {
+			outputFile = "./nitric-spec.json"
+		}
+
+		spec, err := collector.ServiceRequirementsToSpec(proj.Name, envVariables, serviceRequirements, defaultImageName)
+		tui.CheckErr(err)
+
+		marshaler := protojson.MarshalOptions{
+			Multiline: true,
+			Indent:    "  ",
+		}
+
+		specJson, err := marshaler.Marshal(spec)
+		tui.CheckErr(err)
+
+		// output the spec
+		err = os.WriteFile(outputFile, specJson, 0644)
+		tui.CheckErr(err)
+	},
+	Aliases: []string{"spec"},
+}
+
+func init() {
+
+	specCmd.Flags().StringVarP(&debugEnvFile, "env-file", "e", "", "--env-file config/.my-env")
+	specCmd.Flags().StringVarP(&debugFile, "file", "f", "", "--file my-example-spec.json")
```

Comment:
Maybe, `-o` or `--output` is a more expected behaviour.
```suggestion
	specCmd.Flags().StringVarP(&debugFile, "output", "o", "", "--file my-example-spec.json")
```